adage/analyze/management/commands/import_gene_sample_expr.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+"""
+This management command reads an input file of gene-sample expression values
+and loads the valid data into database.  It should be invoked like this:
+
+  python manage.py import_gene_sample_expr <expression_filename>
+
+If a data source (on the first row) or gene name (on the first column) is
+not found in the database, a warning message will be generated and the
+corresponding column or row will be skipped.
+"""
+
+
+from __future__ import print_function
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from genes.models import Gene
+from analyze.models import Sample, ExpressionValue
+
+import logging
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+
+class Command(BaseCommand):
+    help = ("Import gene-sample expression values from an input file.")
+
+    def add_arguments(self, parser):
+        parser.add_argument('expression_filename', type=file)
+
+    def handle(self, **options):
+        try:
+            import_expr(options['expression_filename'])
+            self.stdout.write(self.style.NOTICE(
+                "Imported gene-sample expression values successfully"))
+        except Exception as e:
+            raise CommandError(
+                "Raised exception when importing gene-sample expression "
+                "values: %s" % e)
+
+
+def import_expr(file_handle):
+    """
+    Function that reads input file and load gene-sample expression values
+    into the database.
+    """
+
+    # Enclose reading/importing process in a transaction context manager.
+    # Any exception raised inside the manager will terminate the transaction
+    # and roll back the database.
+    with transaction.atomic():
+        samples = []
+        for line_index, line in enumerate(file_handle):
+            tokens = line.rstrip('\r\n').split('\t')
```

Comment:
Why are we specifying ``'\r\n'`` in ``rstrip()`` here? Is it so that rstrip will not remove single spaces and tabs?